src/app/autocomplete/autocomplete.component.ts

```diff
@@ -1,62 +1,56 @@
-import { Component, OnInit, ElementRef, EventEmitter, Output } from '@angular/core';
-import { WorkshopRepository } from '../services/workshops/workshopRepository'
+import { Component, OnInit, EventEmitter, Output } from '@angular/core';
+import { WorkshopRepository, ILocation } from '../services/workshops/workshopRepository'
 
 @Component({
     selector: 'autocomplete',
-    host: { '(document:click)': 'handleClick($event)' },
     templateUrl: './autocomplete.component.html',
     styleUrls: ['./autocomplete.component.scss']
 })
 export class AutocompleteComponent {
 
     public query = '';
-    public countries = [];
-    public filteredList = [];
-    public elementRef;
+    public countries: ILocation[];
     private workshopRepository: WorkshopRepository;
 
     @Output() selectionChanged = new EventEmitter();
 
-    constructor(myElement: ElementRef, workshopRepo: WorkshopRepository) {
-        this.elementRef = myElement;
+    constructor(private workshopRepo: WorkshopRepository) {
         this.workshopRepository = workshopRepo;
     }
 
     ngOnInit() {
-        this.workshopRepository.getLocations().then(loc => {
-            for (var i = 0; i < loc.length; i++) {
-                this.countries.push(loc[i].name);
-            }
+        this.countries = [];
+        this.workshopRepository.getLocations().then((loc: ILocation[]) => {
+            this.countries = loc;
         });
     }
 
-    filter() {
-        if (this.query !== "") {
-            this.filteredList = this.countries.filter(function (el) {
-                return el.toLowerCase().indexOf(this.query.toLowerCase()) > -1;
-            }.bind(this));
-        } else {
-            this.filteredList = [];
+    public select(value: any) {
```

Comment:
why any?

Reply:
because we need to create interfaces to receive data from 2 places. I'll add a todo.